server/pulp/server/webservices/views/search.py

```diff
@@ -0,0 +1,118 @@
+                               a model that has not yet been converted to MongoEngine. It
+                               should represent an instance of the Model's manager class,
+                               which must have a find_by_criteria() method.
+    :vartype manager:          object
+    :cvar    model:            Define this class attribute if you are making a SearchView for
+                               a MongoEngine Document. The model must have a meta class
+                               attribute defined with the 'queryset_class' key indexing
+                               pulp.server.db.model.base.CriteriaQuerySet, so that the
+                               Document's QuerySet will have a find_by_criteria() method.
+    :vartype model:            mongoengine.Document
+    :cvar    serializer:       If your view needs to modify the QuerySet results before they
+                               are returned to the caller, you can define this attribute. It
+                               should be set to a function that accepts a single object, and
+                               returns a single object.
+    :vartype serializer:       staticmethod
+    """
+    response_builder = staticmethod(util.generate_json_response)
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        Search for objects using an HTTP GET request.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :return:        HttpReponse containing a list of objects that were matched by the request
+        :rtype:         django.http.HttpResponse
+        """
+        query = dict(request.GET)
+
+        fields = query.pop('field', '')
+        if fields:
+            query['fields'] = fields
+
+        return self._generate_response(query)
+
+    @auth_required(authorization.READ)
+    @util.json_body_required
+    def post(self, request):
+        """
+        Search for objects using an HTTP POST request.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :return:        HttpReponse containing a list of objects that were matched by the request
+        :rtype:         django.http.HttpResponse
+        """
+        try:
+            # Retrieve the criteria from the POST data
+            query = request.body_as_json['criteria']
+        except KeyError:
+            raise exceptions.MissingValue(['criteria'])
+
+        return self._generate_response(query)
+
+    @classmethod
+    def _generate_response(cls, query):
+        """
+        Perform the database query using the given search data, and return the resuls as a JSON
+        serialized HttpReponse object.
+
+        :param query: The criteria that should be used to search for objects
+        :type  query: dict
+        :return:      The serialized search results in an HttpReponse
```

Comment:
Not required, but I think a newline in between the inputs and outputs is good. The newline here would be below `:type:` and above `:return:`. I also think the spacing here is very nonstandard with extra spaces randomly put places.